nix/cardano-services/deployments.nix

```diff
@@ -0,0 +1,57 @@
+let
+  inherit (inputs.std) dmerge;
+  inherit (inputs.std.inputs) haumea;
+  inherit (inputs.std.lib.ops) readYAML;
+
+  loadYaml = _: _: readYAML;
+
+  baseline = with haumea.lib;
+    load {
+      src = ./deployments;
+      loader = [(matchers.regex ''^.+\.(yaml|yml)'' loadYaml)];
+    };
+in {
+  dev-preview = dmerge baseline {
+    meta.description = "Development Environment on the Cardano Preview Chain (Frankfurt)";
+    backend-deployment = {
+      spec.template.spec.containers = dmerge.updateOn "name" [
```

Comment:
My big questions going forward here are:
- How will we manage this code as we move to more complex overrides?
- What are the boundaries between dmerge and Kustomize, if both are going to be used together?

Right now this is effectively reimplementing Kustomize functionality, but without the useful domain-specific abstractions for common Kubernetes use cases, like setting a container image version, namespace, or applying common labels.

Reply:
> What are the boundaries between dmerge and Kustomize, if both are going to be used together?

This is easy. If the nix-plane is the source of a configuration datum, use it to ensure cheap dynamic value updates.  If not, keep that datum self-contained "within the YAML" (including `Kustomization` for ease of maintenance). If a value is overwhelmingly a constant, but technically emanates from the nix-plane, then make an educated judgement with propensity to prefer Kustomization over nix-plane sourcing of that nearly constant configuration datum.

> How will we manage this code as we move to more complex overrides?

I don't expect more than one layer of overrides in this scenario, especially since we can leverage Kustomize for many things. If, after gathering more feedback, we conclude that multi-layer / nested overlays are unavoidable we can do so by duck-chaining overlays with the `dmerge.chainable` facility that was design precisely for more complex overlays.

But I can't yet fathom that this may be even necessary, given how much of kubernete's accidential complexity due to flat data structures is already mitigated by `Kustomization`.